pkg/daemon/cleanup/cleanup.go

```diff
@@ -72,47 +96,55 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 	return nil
 }
 
-// getServicesToRemove returns a list of services that should be removed as part of the cleanup process
-// returns the merged Data of the latest ConfigMap, and the ConfigMap specified by the node's version annotation
-func getServicesToRemove(ctx context.Context, cli client.Client, node *core.Node, configMapNamespace string) ([]servicescm.Service, error) {
-	// get data from the latest services ConfigMap
-	latestCM, err := servicescm.GetLatest(cli, ctx, configMapNamespace)
+// ensureEnvVarsAreRemoved removes all WICD configured ENV variables from this instance
+func ensureEnvVarsAreRemoved(cm *core.ConfigMap) error {
```

Comment:
Ideally we should be following the same pattern with services i.e. get the env variables to remove and  then remove them. I also don't fully understand the split between `ensureEnvVarsAreRemoved()` and `controller.EnsureEnvVarsAreRemoved() `. To be clear I am not asking for any changes here but just finding the logic split of methods a bit hard to follow across packages.

Reply:
Thanks for the feedback, I ll give it a try if I can make it less complex. I am also working on merging the CM's through a common function, I'll add it to a separate commit for easier review.

Reply:
I don't think we should add merging of the environment if it's too complex. Plus we don't want to expand the scope of the bug.

Reply:
This function doesn't make it clear that it will restart the computer if theres any change.
Please rename this to reflect what it is doing. `removeAllWatchedEnvVarsAndRestart`

Reply:
I am refactoring this function it will now only return if restart is required and not perform it.